src/libsyncengine/requests/serverrequests.cpp

```diff
@@ -985,6 +986,120 @@ ExitCode ServerRequests::getUserFromSyncDbId(int syncDbId, User &user) {
+    std::function<bool(char, int)> safeProgressCallback =
+        progressCallback != nullptr ? std::function<bool(char, int)>([progressCallback](char status, int percent) { return progressCallback(status, percent); })
+                                    : std::function<bool(char, int)>([](char, int) { return true; });
+    
+    safeProgressCallback('A', 0);
+
+    if (bool found = false; !ParmsDb::instance()->updateAppState(AppStateKey::LastLogUploadArchivePath, "", found) || !found) {
+        LOG_WARN(Log::instance()->getLogger(), "Error in ParmsDb::updateAppState");
+        // Do not return here because it is not a critical error, especially in this context where we are trying to send logs
+    }
+
+    const SyncPath logUploadTempFolder = CommonUtility::getAppSupportDir() / "logUploadTemp";
+    IoError ioError = IoErrorSuccess;
+    IoHelper::createDirectory(logUploadTempFolder, ioError);
+    if (ioError == IoErrorDirectoryExists) {  // If the directory already exists, we delete it and recreate it
+        IoHelper::deleteDirectory(logUploadTempFolder, ioError);
+        IoHelper::createDirectory(logUploadTempFolder, ioError);
+    }
+
+    if (ioError != IoErrorSuccess) {
+        LOG_WARN(Log::instance()->getLogger(),
+                 "Error in IoHelper::createDirectory : " << Utility::formatIoError(logUploadTempFolder, ioError).c_str());
+        exitCause = ioError == IoErrorDiskFull ? ExitCauseNotEnoughDiskSpace : ExitCauseUnknown;
+        return ExitCodeSystemError;
+    }
+
+    std::function<bool(int)> progressCallbackArchivingWrapper = [&safeProgressCallback](int percent) {
+        return safeProgressCallback('A', percent);
+    };
+
+
+    SyncPath archivePath;
+    exitCode = LogArchiver::generateLogsSupportArchive(includeArchivedLog, logUploadTempFolder, progressCallbackArchivingWrapper,
+                                                       archivePath, exitCause);
+    if (exitCode != ExitCodeOk) {
+        LOG_WARN(Log::instance()->getLogger(),
+                 "Error in LogArchiver::generateLogsSupportArchive: " << exitCode << " : " << exitCause);
+        IoHelper::deleteDirectory(logUploadTempFolder, ioError);
+        return exitCode;
+    }
+
+    if (bool found = false;
+        !ParmsDb::instance()->updateAppState(AppStateKey::LastLogUploadArchivePath, archivePath.string(), found) || !found) {
+        LOG_WARN(Log::instance()->getLogger(), "Error in ParmsDb::updateAppState");
+    }
+
+    // Upload archive
+    std::function<bool(int)> progressCallbackUploadingWrapper = [&safeProgressCallback](int percent) {
+        return safeProgressCallback('U', percent);
+    };
+
+    for (int i = 0; i < 100; i++) {  // TODO: Remove | Fake progress waiting for the real upload implementation
+        if (!progressCallbackUploadingWrapper(i)) {
+            exitCode = ExitCodeOperationCanceled;
+            break;
+        }
+        Utility::msleep(100);
+    }
+
+
+    // TODO: implement real log upload backend
```

Comment:
So it is not yet the definitive implementation? Should we really merge into `develop`? Waht if we need to release a new version before the final implementation?

Reply:
As discussed, this PR includes only the backend for log uploads. The functions are currently not visible to the user.